WireUI/Sources/WireReusableUIComponents/LabeledTextField/LabeledTextField.swift

```diff
@@ -45,38 +52,112 @@ public struct LabeledTextField: View {
                             .foregroundColor(ColorTheme.Base.requiredField.color)
                     ) : Text(title)
                 )
-                .wireTextStyle(.h4)
+                .foregroundStyle(titleColor(isEnabled: isEnabled, isFocused: isFocused))
+                .wireTextStyle(.subline1)
+            }
+            HStack(spacing: 0) {
+                TextField(placeholder ?? "", text: $string)
+                    .wireTextStyle(.body1)
+                    .focused($isFocused)
+                    .foregroundStyle(labelColor(isEnabled: isEnabled))
+                    .padding(.vertical, 12)
+                if !string.isEmpty, isEnabled {
+                    Button(action: {
+                        string = ""
+                    }, label: {
+                        Image(systemName: "xmark.circle.fill")
+                            .foregroundStyle(.black)
+                            .frame(width: 16, height: 16)
+                            .padding(19)
+                    })
+                }
+            }
+            .padding(.leading, 16)
+            .background {
+                if #available(iOS 17.0, *) {
+                    RoundedRectangle(cornerRadius: 12)
+                        .fill(labelBackgroundColor(isEnabled: isEnabled))
+                        .stroke(labelBorderColor(isEnabled: isEnabled, isFocused: isFocused), lineWidth: 1)
+                } else {
+                    RoundedRectangle(cornerRadius: 12)
+                        .stroke(labelBorderColor(isEnabled: isEnabled, isFocused: isFocused), lineWidth: 1)
+                        .background(labelBackgroundColor(isEnabled: isEnabled))
+                        .cornerRadius(12)
+                }
             }
-            TextField(placeholder ?? "", text: $string)
-                .textFieldStyle(.roundedBorder)
-                .wireTextStyle(.body1)
         }
     }
 }
 
+private func titleColor(isEnabled: Bool, isFocused: Bool) -> Color {
+    if isEnabled, isFocused {
+        return ColorTheme.Base.onPrimaryVariant.color
+    }
+    if isEnabled {
+        return ColorTheme.Base.labelTitle.color
+    }
+    return ColorTheme.Base.labelTitle.color
+}
+
+private func labelColor(isEnabled: Bool) -> Color {
+    if isEnabled {
+        return .primaryText
+    }
+    return ColorTheme.Base.onDisabled.color
+}
+
+private func labelBackgroundColor(isEnabled: Bool) -> Color {
+    if isEnabled {
+        return .clear
+    }
+    return ColorTheme.Backgrounds.background.color
+}
+
+private func labelBorderColor(isEnabled: Bool, isFocused: Bool) -> Color {
+    if isEnabled, isFocused {
+        return ColorTheme.Base.onPrimaryVariant.color
+    }
+    if isEnabled {
+        return ColorTheme.Strokes.outline.color
+    }
+    return ColorTheme.Strokes.outline.color
+}
```

Comment:
question: is it reused somewhere? otherwise maybe better to move it within the LabeledTextField, no?